rep-2004.rst

```diff
@@ -0,0 +1,521 @@
+    * All pull requests will be tested via CI, and on all tier 1 platforms (if applicable)
+    * Any required changes to documentation (API documentation, feature documentation, release notes, etc.) must be proposed before merging related changes
+
+* Documentation:
+
+  * Must have documentation for each "feature" (e.g. for ``rclcpp``: create a node, publish a message, spin, etc.)
+  * Must have documentation for each item in the public API (e.g. functions, classes, etc.)
+  * Must have a declared license or set of licenses
+  * Must have a copyright statement in each source file
+  * Must have a "quality declaration" document, which declares the quality level and justifies how the package meets each of the requirements
+
+    * Must have a section in the repository's ``README`` which contains the "quality declaration" or links to it
+    * Must register with a centralized list of 'Level 1' packages, if one exists, to allow for peer review of the claim
+
+  * [ROS Core]:
+
+    * Must have automated checks for copyright statements and licenses
+    * Must use the Apache 2.0 license, unless the package has an existing permissive license (e.g. rviz uses three-clause BSD)
+
+* Testing:
+
+  * Must have system tests which cover all items in the "feature" documentation
+  * Must have system, integration, and/or unit tests which cover all of the public API
+  * Code coverage:
+
+    * Must have code coverage tracking for the package
+    * Must have and enforce a code coverage policy for new changes
+    * [ROS Core]:
+
+      * Must provide line coverage
+      * Must achieve a line coverage above 95%
+      * May pick a lower percentage target with justification, but must document it prominently
+      * May provide branch coverage
+      * May exclude code from coverage (test code, debug code, etc.)
+      * Must require coverage to increase or stay the same before merging a change, but...
+      * May accept a change that decreases coverage with proper justification (e.g. deleting code that was previously covered can cause the percentage to drop)
+
+  * Performance:
+
+    * Must have performance tests (exceptions allowed if they don't make sense to have)
+    * Must have a performance regression policy (i.e. blocking either changes or releases on unexpected performance regressions)
+    * [ROS Core]:
+
+      * May have performance tests, strongly recommended, but for some packages it doesn't make sense
+      * If there are performance tests, must choose to either check each change or before each release or both
+      * If there are performance tests, must require justification for merging a change or making a release that lowers performance
+
+  * Linters and Static Analysis
+
+    * Must have a code style and enforce it.
+    * Must use static analysis tools where applicable.
+    * [ROS Core]:
+
+      * Must use ROS code style and use linters from ``ament_lint_common`` to enforce it
```

Comment:
IMHO, `clang-tidy` should be one of the linters in `ament_lint_common`.

Reply:
Perhaps so, but for me that's an issue to be filed with `ament_lint_common`, and should stay out of this REP (since the list of linters and analyzers may change over time and updating the REP is tedious.